### PR TITLE
feat: add typescript to connect, mapStateToProps, mapDispatchToProps

### PR DESCRIPTION
**Pull request by nikitsch**

### CODE REVIEW

File: src/components/Profile/ProfileInfo/ProfileStatus.tsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  return users.filter(u => true);
+})
+
+// export const getUsers = (state) => {
+//   return state.usersPage.users;
+// }
+
+// export const getUsersSuperSelector = createSelector(getUsers, getIsFetching, (users, isFetching) => {
+//   return users.filter(u => true);
+// })
```
Comment (GolDOragon) [inline src/components/Profile/ProfileInfo/ProfileStatus.tsx:18]: remove

File: src/components/Users/Users.tsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
+import Pagination from '../common/Pagination/Pagination';
+import User from './User';
+import styles from './Users.module.css';
+import { UserType } from '../../types/types';
+import React, { FC } from 'react';
+
+type PropsType = {
```
Comment (GolDOragon) [inline src/components/Users/Users.tsx:7]: `type UserProps = {`

File: src/components/Users/Users.tsx [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  totalUsersCount: number
+  pageSize: number
+  users: Array<UserType>
+  followingInProgress: Array<number>
+  unfollow: (userId: number) => void
+  follow: (userId: number) => void
+}
+
+let Users: FC<PropsType> = ({ currentPage, onPageChanged, totalUsersCount, pageSize, users, ...props }) => {
+  return <div>
```
Comment (GolDOragon) [inline src/components/Users/Users.tsx:19]: `const User: FC<UserProps> =`

File: src/components/Users/UsersContainer.tsx [hunk 4]
```diff
@@ -6,20 +6,44 @@ import Preloader from '../common/Preloader/Preloader';
 // import { withAuthRedirect } from '../../hoc/withAuthRedirect';
```
Comment (GolDOragon) [inline src/components/Users/UsersContainer.tsx:6]: remove

File: src/components/common/Pagination/Pagination.tsx [hunk 5]
```diff
@@ -1,12 +1,20 @@
 import React, { useState } from 'react';
 import styles from './Pagination.module.css';
 import cn from 'classnames';
+import { FC } from 'react';
```
Comment (GolDOragon) [inline src/components/common/Pagination/Pagination.tsx:4]: `import React, { FC, useState } from 'react';`
remove duplicated import

File: src/components/common/Pagination/Pagination.tsx [hunk 6]
```diff
@@ -1,12 +1,20 @@
...
 
-let Pagination = ({ currentPage, onPageChanged, totalItemsCount, portionSize = 15 }) => {
+type PropsType = {
+  currentPage: number
+  onPageChanged: (pageNumber: number) => void
+  totalItemsCount: number
+  portionSize?: number
+}
+
+let Pagination: FC<PropsType> = ({ currentPage, onPageChanged, totalItemsCount, portionSize = 15 }) => {
```
Comment (GolDOragon) [inline src/components/common/Pagination/Pagination.tsx:13]:
```
type PaginationProps = { ... }

const Pagination: FC<PaginationProps> = 
```

File: src/components/common/Pagination/Pagination.tsx [hunk 7]
```diff
@@ -1,12 +1,20 @@
...
 
   let pagesCount = Math.ceil(totalItemsCount / portionSize);
 
-  let pages = [];
+  let pages: Array<number> = [];
   for (let i = 1; i <= pagesCount; i++) {
```
Comment (GolDOragon) [inline src/components/common/Pagination/Pagination.tsx:18]: const pages: Array<number> = [];

File: src/redux/redux-store.ts [hunk 8]
```diff
@@ -18,10 +18,14 @@ let reducers = combineReducers ({
...
+// @ts-ignore
 const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
-const store = createStore(reducers, composeEnhancers(applyMiddleware(thunkMiddleware)));
+const store = createStore(rootReducer, composeEnhancers(applyMiddleware(thunkMiddleware)));
 
-// let store = createStore(reducers, applyMiddleware(thunkMiddleware));
+// let store = createStore(rootReducer, applyMiddleware(thunkMiddleware));
 
 // TODO: Удалить, но потом
 // window.store = store;
```
Comment (GolDOragon) [inline src/redux/redux-store.ts:31]: remove!

File: src/redux/redux-store.ts [hunk 9]
```diff
@@ -18,10 +18,14 @@ let reducers = combineReducers ({
   app: appReducer
 });
 
+type RootReducerType = typeof rootReducer;
+export type AppStateType = ReturnType<RootReducerType>
+
+// @ts-ignore
 const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
```
Comment (GolDOragon) [inline src/redux/redux-store.ts:25]: Need to add global.d.ts in project. https://stackoverflow.com/a/52801110/19804311

```ts
declare global {
    interface Window {
      __REDUX_DEVTOOLS_EXTENSION_COMPOSE__?: typeof compose;
    }
}
const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
```

File: src/redux/users-reducer.ts [hunk 10]
```diff
@@ -17,18 +17,13 @@ const TOGGLE_IS_FOLLOWING_PROGRESS = 'TOGGLE_IS_FOLLOWING_PROGRESS';
...
-  pageSize: 20 as number | null,
-  totalUsersCount: 0 as number | null,
-  currentPage: 1 as number | null,
-  isFetching: true as boolean | null,
-  followingInProgress: [] as Array<FollowingInProgressType>
+  pageSize: 20, // as number | null или as number
+  totalUsersCount: 0, // as number | null,
+  currentPage: 1, // as number | null,
+  isFetching: true, // as boolean | null,
+  followingInProgress: [] as Array<number>
```
Comment (GolDOragon) [inline src/redux/users-reducer.ts:26]: remove unused code

File: src/redux/users-reducer.ts [hunk 11]
```diff
@@ -17,18 +17,13 @@ const TOGGLE_IS_FOLLOWING_PROGRESS = 'TOGGLE_IS_FOLLOWING_PROGRESS';
 //   error: string
 // }
```
Comment (GolDOragon) [inline src/redux/users-reducer.ts:19]: also remove

File: src/redux/users-selectors.ts [hunk 12]
```diff
@@ -16,22 +17,22 @@ export const getUsers = createSelector(getUsersSelector, (users) => {
 //   return users.filter(u => true);
 // })
```
Comment (GolDOragon) [inline src/redux/users-selectors.ts:18]: remove

**Review by GolDOragon — CHANGES_REQUESTED** [review]

File: src/components/Profile/ProfileInfo/ProfileStatus.tsx (hunk 1 above)
Comment (nikitsch) [reply src/components/Profile/ProfileInfo/ProfileStatus.tsx:18]: For educational purposes I will leave this code

File: src/redux/redux-store.ts (hunk 8 above)
Comment (nikitsch) [reply src/redux/redux-store.ts:31]: For educational purposes I will leave this code

File: src/redux/users-reducer.ts (hunk 10 above)
Comment (nikitsch) [reply src/redux/users-reducer.ts:26]: For educational purposes I will leave this code

File: src/components/Users/Users.tsx [hunk 13]
```diff
@@ -0,0 +1,38 @@
+import Pagination from '../common/Pagination/Pagination';
+import User from './User';
+import styles from './Users.module.css';
+import { UserType } from '../../types/types';
+import React, { FC } from 'react';
+
+type UsersPropsType = {
+  currentPage: number
```
Comment (GolDOragon) [inline src/components/Users/Users.tsx:8]: UsersProps not UsersPropsType

File: src/components/Users/Users.tsx [hunk 14]
```diff
@@ -0,0 +1,38 @@
...
+  onPageChanged: (pageNumber: number) => void
+  totalUsersCount: number
+  pageSize: number
+  users: Array<UserType>
+  followingInProgress: Array<number>
+  unfollow: (userId: number) => void
+  follow: (userId: number) => void
+}
+
+let Users: FC<UsersPropsType> = ({ currentPage, onPageChanged, totalUsersCount, pageSize, users, ...props }) => {
```
Comment (GolDOragon) [inline src/components/Users/Users.tsx:18]: const Users

**Review by GolDOragon — APPROVED** [review]

File: src/components/common/Pagination/Pagination.tsx (hunk 7 above)
Comment (nikitsch) [reply src/components/common/Pagination/Pagination.tsx:17]: you probably meant:
const pages: number[] = [];

**Pull request merged**